shanoir-ng-front/src/app/shared/components/table/table.component.ts

```diff
@@ -121,6 +124,32 @@ export class TableComponent implements OnInit, OnChanges, OnDestroy {
         });
     }
 
+    newTabRouting(col, item): string {
+      this.path = this.router.url;
+
+      if (!this.editMode) {
+        // case of study-list, subject-list, examination-list, datasetacquisition-list
+        if (col.type != "button" && this.router.url.endsWith('list') ) {
+          this.path = this.path.replace("list", "");
+          return this.path + 'details/'+ item.id
+        }
+        // case of solr search
+        else if (this.router.url.endsWith('solr-search')) {
```

Comment:
I'm not ok with using this.router.url.endsWith('solr-search'), or any this.router.url.endsWith. The table component should be unaware of any "solr" or "dataset" things. Intead of all those if else you should find a way to make the component that is calling <table> responsible for the building of this link.

I see 2 possibilities : 
* use the ready made ```route?: (item: any) => string``` provided in ColumnDefinition (column.definition.type.ts)
* add a detailsUrl property in ColumnDefinition and build the url with it + item.id (if exists)